webhook_test.go

```diff
@@ -84,6 +85,131 @@ func TestLoadInitFile(t *testing.T) {
 // 	t.Logf("UUID: %v", uuid)
 //}
 
+func TestIsSafe(t *testing.T) {
+	// test 1 Create OK
+	tpod := getFakePod("tests/FakeIsSafeCreateOK.json")
+	safe, err := isSafe(&tpod, "CREATE")
+	if safe && err.Error() == "" {
+		t.Logf("Testing CREATE with safe pod successful")
+	} else {
+		t.Error("Testing CREATE with safe pod failed")
+	}
+
+	// test 2 Create Error
+	tpod = getFakePod("tests/FakeIsSafeCreateError.json")
+	safe, err = isSafe(&tpod, "CREATE")
+	if !safe && err.Error() == "Using hostPath Volume with '/tsi-secure/sockets' is not allowed" {
+		t.Logf("Testing CREATE with non-safe pod successful")
+	} else {
+		t.Error("Testing CREATE with non-safe pod failed")
+	}
+
+	// test 3 Update OK
+	tpod = getFakePod("tests/FakeIsSafeUpdateOK.json")
+	safe, err = isSafe(&tpod, "UPDATE")
+	if safe && err.Error() == "" {
+		t.Logf("Testing UPDATE with safe pod successful")
+	} else {
+		t.Error("Testing UPDATE with safe pod failed")
+	}
+
+	// test 4 Update Error
+	tpod = getFakePod("tests/FakeIsSafeUpdateError.json")
+	safe, err = isSafe(&tpod, "UPDATE")
+	if !safe && err.Error() == "Attempt to modify the sidecar image to ubuntu:latest" {
```

Comment:
If specific errors are required to check errcodes, check out the use of https://golang.org/pkg/errors/#Is, and errors should be defined as variables.